### PR TITLE
Remove lib.d.ts, update README.md, add CHANGELOG.md and LICENSE

### PR DESCRIPTION
**Pull request by perludum**
As of TypeScript 2.0 the standard library declarations (ES5 to ES2017) are built in, so I'd propose removing the lib.d.ts from the project - also in order to allow other standard libs (like ES3).

I updated the README.md and added some notes about ECMAScript 3 and how to deal with a suitable polyfill for ECMAScript 5.

Also moved the version notes to a CHANGELOG.md and added a LICENSE file.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,62 +1,109 @@
...
+# Photoshop.d.ts
 
 [![npm version](https://badge.fury.io/js/photoshop.d.ts.svg)](http://badge.fury.io/js/photoshop.d.ts)
 [![Build Status](https://travis-ci.org/felixSchl/photoshop.d.ts.svg)](https://travis-ci.org/felixSchl/photoshop.d.ts)
 ![dependencies status](https://david-dm.org/felixschl/photoshop.d.ts.svg)
 
-> Type-safe Photoshop scripting
+**Type-safe Photoshop scripting**
+
+This project aims to create and maintain TypeScript declaration files for Adobe ExtendScript, focussed on Photoshop. It enables you writing Photoshop scripts with the power and safety of TypeScript.
```
Comment (felixSchl) [inline README.md]: A nit pick, but let's keep these lines under 80 characters so it looks good on the terminal when viewed via `less` and for those of us using command line editors... Also: "It enables you writing Photoshop scripts" should probably read "It enables you to write Photoshop scripts"

File: README.md [hunk 2]
```diff
@@ -1,62 +1,109 @@
...
-The typescript declarations are generated from the official javascript reference
-PDFs. The text is copied and pasted into local files. The **parser** will parse
-the raw text file and extract meaningful information from the respective
-chapters of the documentation. Once the **parser** has delivered a in-memory
-representation of the types and constants, the **emitter** will render types and
-constants as typescript files.
+1. Download es5.js and place it into your project folder.
+2. Assure `checkJS` ist set to `false` in your `tsconfig.json` when using a 
+JS polyfill library.
+2. Link the polyfill in the `files` section of your `tsconfig.json`:
```
Comment (felixSchl) [inline README.md]: Either increment this to "3." or make all of them "1." to get automatic numbering

**Comment by felixSchl** [comment]
You should add your name to the Copyright file as well. Full disclosure: I have not much of a clue about licensing, but it seems if you do this much work you should have your name in there

File: README.md [hunk 3]
```diff
@@ -1,62 +1,109 @@
...
+
+Features:
+
+* TypeScript type safety for ExtendScript projects
+* View ExtendScript object documentation in-place for various code editors
+* Live type checking, code completion, parameter info, quick info, and member lists for ExtendScript objects with e.g. IntelliSense
+
+Demo:
+
+![Alt text](demo.gif?raw=true "Title")
```
Comment (felixSchl) [inline README.md]: Neat trick, did not know of this one! :)

**Comment by perludum** [comment]
Thank you for your review!
There's also a problem with travis and I will investigate it shortly.

**Comment by perludum** [comment]
OK I got it with travis: The declarations files in this project need to be tested with slightly different conditions (in particular just with the ES5 standard lib) than the app.ts. I fixed this in the package.json. So there will be free choice of the standard lib now :)
I tested the declarations again in a TS/ES project with the new setup and I think it's ready to merge now.

Thank you for your notes, they are merged in with the latest readme.md commit!

**Comment by felixSchl** [comment]
Sweet, it looks like this package is coming together quite nicely, do you think it's time for 1.x.x with these updates or do you have further changes in the back of your mind that will happen soon? Would you be able to update changelog.md with these changes, either assuming 0.5.0 or 1.0.0

**Comment by perludum** [comment]
Absolutely, I think this package is ready for a 1.0 release!

Currently I’ve no further changes in my mind, because everything seems to work fine so far. I didn’t test each and every declaration, but in the next weeks, I’ll use this lib actively in an ExtendScript project and see if there are any issues left and would try to fix them, if at all. 

As a last-minute update, the comment format in the ExtendScript declarations now have a line length of 80 characters, too, and it even looks nicer.

Finally, the changelog is updated for a major version release :)

**Pull request merged**